mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/dao/PostgresAttachmentDAO.java

```diff
@@ -80,9 +84,21 @@ public Mono<Pair<AttachmentMetadata, BlobId>> getAttachment(AttachmentId attachm
                 blobIdFactory.from(row.get(PostgresAttachmentTable.BLOB_ID))));
     }
 
+    public Flux<AttachmentMetadata> getAttachments(Collection<AttachmentId> attachmentIds) {
+        return postgresExecutor.executeRows(dslContext -> Flux.from(dslContext.selectFrom(PostgresAttachmentTable.TABLE_NAME)
+                .where(PostgresAttachmentTable.ID_AS_UUID.in(attachmentIds.stream().map(AttachmentId::asUUID).collect(ImmutableList.toImmutableList())))))
```

Comment:
1. Why do we don't use directly "AttachmentId.getId()" in `where` clause?
2. When using "IN" clause, we should check the size of IN, if large, we should partition it, see the same `PostgresMailboxMessageDAO#findMessagesByMailboxIdAndUIDs`

Reply:
(1) updated
Understood why you need ID and ID_AS_UUID
I think we can use a single column for it. (`ID`, maybe. And change type from UUID -> VARCHAR)
it is enough.

Reply:
Removed ID_AS_UUID. Use ID as pk (change type from UUID to VARCHAR)